### PR TITLE
feat: Dockerize dotnet services

### PR DESCRIPTION
**Pull request by pkanal**
## Which problem is this PR solving?
This PR dockerizes the .NET services and changes the Tiltfile to run .NET services through Docker.

## Short description of the changes
- Update docker-compose.yml for .NET services to be able to run on its own
- Updated Tiltfile to run .NET services with the docker compose

## How to test
1. Pull the changes locally
2. Run `tilt up dotnet` from the root directory
3. Check that the greeting service works

Things to test:
- [x] `tilt up dotnet` works from the root directory
- [x] `docker-compose up` works from the `dotnet` directory
- [x] Other tilt commands work for services in other languages (e.g. `tilt up go`)

### CODE REVIEW

**Review by cartermp — APPROVED** [review]
Ooo, some nice TLC

File: dotnet/README.md [hunk 1]
```diff
@@ -9,4 +9,8 @@ You'll note that instrumentation is done differently depending on the project:
 * `year-service` is a service built as a minimal .NET 6 API that uses the standard OpenTelemetry libraries to configure instrumentation. It's more lines of code.
 * `name-service` is a service built with .NET 5 that uses the standard OpenTelemetry libraries.
 
-The goal here is to demonstrate that you can configure instrumentation in several ways with .NET but still send data to Honeycomb.
\ No newline at end of file
+The goal here is to demonstrate that you can configure instrumentation in several ways with .NET but still send data to Honeycomb.
+
+### Run the services
```
Comment (JamieDanielson) [inline dotnet/README.md:14]: There's a dotnet section in `DEVELOPING.md` that can probably change to match node, where other details are removed and it just says to view the language-specific setup instructions in the readme.

Also, maybe we should add to `tilt down dotnet` twice to kill the services?

**Review by JamieDanielson — APPROVED** [review]
that's awesome

**Review by JamieDanielson — COMMENTED** [review]
Oh, hm. Actually I couldn't get this to work by using `docker-compose` directly from the directory, because it fails to talk to Redis 😢 
`StackExchange.Redis.RedisConnectionException: No connection is active/available to service this operation: SRANDMEMBER messages; UnableToConnect on redis:6379/Interactive,...`

We may need to keep redis in this docker compose after all instead of only loading from the root...

**Comment by vreynolds** [comment]
superseded by #608

**Pull request closed without merging**